### PR TITLE
refactor: improve eslint

### PR DESCRIPTION
**Pull request by septs**
closes #

### CODE REVIEW

File: packages/maskbook/src/utils/hooks/useQRCodeVideoScan.tsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ import '../../components/QRScanner/ShapeDetectionPolyfill'
 
 export async function getBackVideoDeviceId() {
     const devices = filter(await navigator.mediaDevices.enumerateDevices(), ({ kind }) => kind === 'videoinput')
-    const back = find(devices, ({ label }) => !/Front/i.test(label) && /Back|Rear/i.test(label))
+    const back = find(devices, ({ label }) => !/front/i.test(label) && /back|rear/i.test(label))
```
Comment (Jack-Works) [inline packages/maskbook/src/utils/hooks/useQRCodeVideoScan.tsx:11]: Is this change OK?

File: packages/maskbook/src/social-network-adaptor/twitter.com/automation/pasteTextToComposition.ts [hunk 2]
```diff
@@ -53,5 +53,5 @@ export const pasteTextToCompositionTwitter: SocialNetworkUI.AutomationCapabiliti
         setTimeout(() => {
             abortCtr.abort()
         }, timeout)
-        worker(abortCtr).then(undefined, (e) => fail(e))
+        worker(abortCtr).then(undefined, (error) => fail(error))
```
Comment (Jack-Works) [inline packages/maskbook/src/social-network-adaptor/twitter.com/automation/pasteTextToComposition.ts:56]: I think `e` is also an acceptable name

File: packages/maskbook/src/social-network-adaptor/twitter.com/encoding.ts [hunk 3]
```diff
@@ -68,7 +68,7 @@ export const twitterEncoding = {
             return [
                 `🎼${batchReplace(
                     payload
-                        .replace(/^PostData_v\d=/i, '')
+                        .replace(/^postdata_v\d=/i, '')
```
Comment (Jack-Works) [inline packages/maskbook/src/social-network-adaptor/twitter.com/encoding.ts:71]: I don't think this change is good. The uppercase makes it easier to read.

File: packages/maskbook/src/plugins/ITO/SNSAdaptor/hooks/useFill.ts [hunk 4]
```diff
@@ -280,8 +280,8 @@ export function useFillParams(poolSettings: PoolSettings | undefined) {
             .estimateGas({
                 from: account,
             })
-            .catch((err: Error) => {
-                gasError = err
+            .catch((error: Error) => {
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/ITO/SNSAdaptor/hooks/useFill.ts:283]: Also, I think `err` is an acceptable name.

File: packages/maskbook/src/utils/hooks/useQRCodeVideoScan.tsx (hunk 1 above)
Comment (septs) [reply packages/maskbook/src/utils/hooks/useQRCodeVideoScan.tsx:11]: OK

File: packages/maskbook/src/social-network-adaptor/twitter.com/encoding.ts (hunk 3 above)
Comment (Jack-Works) [reply packages/maskbook/src/social-network-adaptor/twitter.com/encoding.ts:71]: This one didn't get resolved. This rule seems decrease the readability of the code.

Comment (septs) [reply packages/maskbook/src/social-network-adaptor/twitter.com/encoding.ts:71]: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/better-regex.md

Comment (Jack-Works) [reply packages/maskbook/src/social-network-adaptor/twitter.com/encoding.ts:71]: This link doesn't help with this problem. `PostData` is more readable than `postdata`

File: packages/maskbook/src/plugins/Trader/trader/uniswap/usePairs.ts [hunk 5]
```diff
@@ -45,8 +45,8 @@ export function usePairs(tokenPairs: readonly TokenPair[]) {
                 if (x.error) return undefined
                 return {
                     id: contracts[i].options.address,
-                    reserve0: x.value._reserve0,
-                    reserve1: x.value._reserve1,
+                    reserve0: x.value!._reserve0,
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Trader/trader/uniswap/usePairs.ts:48]: `if (!x.value) return undefined`

File: packages/.eslintrc.json [hunk 6]
```diff
@@ -25,6 +25,15 @@
         "import/no-deprecated": "warn",
         "lodash/import-scope": ["error", "member"],
         "unused-imports/no-unused-imports-ts": "warn",
+        "unicorn/better-regex": "error",
```
Comment (Jack-Works) [inline packages/.eslintrc.json:28]: I'm okay with other rules, but let's wait for the uppercase problem in the RegExp

Comment (septs) [reply packages/.eslintrc.json:28]: i not think the is problem

**Review by nuanyang233 — DISMISSED** [review]

File: packages/.eslintrc.json (hunk 6 above)
Comment (Jack-Works) [reply packages/.eslintrc.json:28]: image

I found this rule doesn't affect the case. I'll revert those changes in the letter case.

**Review by Jack-Works — DISMISSED** [review]

**Comment by Jack-Works** [comment]
https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1476

**Review by Jack-Works — APPROVED** [review]

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**